### PR TITLE
[refactor] remove gpc dependency in colotensor's _ops

### PR DESCRIPTION
**Pull request by feifeibear**

### CODE REVIEW

File: colossalai/nn/_ops/layernorm.py [hunk 1]
```diff
@@ -17,7 +17,8 @@ def colo_layernorm(
     input_tensor, weight, bias = tuple(map(convert_to_colo_tensor, (input_tensor, weight, bias)))
 
     # TODO (ver217): check dist spec
-    input_tensor = input_tensor.convert_to_dist_spec(distspec.replicate(input_tensor.tensor_spec.get_process_group()))
+    input_tensor = input_tensor.convert_to_dist_spec(
+        distspec.replicate(input_tensor.tensor_spec.dist_spec.process_group))
```
Comment (ver217) [inline colossalai/nn/_ops/layernorm.py:21]: Why?

File: colossalai/nn/parallel/data_parallel.py [hunk 2]
```diff
@@ -191,8 +193,11 @@ class ZeroDDP(ColoDDP):
             For more details, see the API reference of ``GeminiManager``.
     """
 
-    def __init__(self, module: torch.nn.Module, gemini_manager: GeminiManager) -> None:
-        super().__init__(module.half())
+    def __init__(self,
+                 module: torch.nn.Module,
+                 gemini_manager: GeminiManager,
+                 process_group: Optional[ColoProcessGroup] = None) -> None:
```
Comment (ver217) [inline colossalai/nn/parallel/data_parallel.py:199]: ColoDDP receives `dist.ProcessGroup`, and ZeroDDP receives `ColoProcessGroup`. Is this difference necessary?

Comment (feifeibear) [reply colossalai/nn/parallel/data_parallel.py:199]: I will update it in the next PR.

File: tests/test_tensor/test_zero_optim.py [hunk 3]
```diff
@@ -79,44 +75,53 @@ def run_gpt(use_chunk, use_zero, placement_policy, tp_init_spec_func=None):
...
+    check_param_equal(model, torch_model, pg)
     model.train()
     torch_model.train()
-    set_seed(gpc.get_local_rank(ParallelMode.DATA))
+    set_seed(pg.dp_local_rank())
     for i, (input_ids, attn_mask) in enumerate(train_dataloader):
         if i > 2:
             break
+
+        torch.distributed.broadcast(input_ids, 0)
```
Comment (ver217) [inline tests/test_tensor/test_zero_optim.py:112]: Don't broadcast. We except DP ranks have different inputs.

Comment (feifeibear) [reply tests/test_tensor/test_zero_optim.py:112]: OK

File: colossalai/nn/parallel/data_parallel.py (hunk 2 above)
Comment (feifeibear) [reply colossalai/nn/parallel/data_parallel.py:199]: updated, both of them use ColoProcessGroup

**Review by ver217 — APPROVED** [review]

**Pull request merged**